be/src/olap/rowset/rowset.cpp

```diff
@@ -32,7 +32,7 @@ Rowset::Rowset(const TabletSchema* schema, const std::string& tablet_path,
         _is_cumulative = version.first != version.second;
     }
     // build schema from RowsetMeta.tablet_schema or Tablet.tablet_schema
-    _schema = _rowset_meta->tablet_schema() != nullptr ? _rowset_meta->tablet_schema() : schema;
+    _schema = _rowset_meta->tablet_schema() ? _rowset_meta->tablet_schema().get() : schema;
```

Comment:
should also change _schema in rowset.h to TabletSchemaSPtr. There maybe segment fault if we add clean logic in tablet schema cache.